### PR TITLE
Make it build with ghc-9.10

### PR DESCRIPTION
**Pull request by erikd**
# Changelog

```yaml
- description: |
    Support ghc-9.10
# uncomment types applicable to the change:
  type:
  - maintenance    # not directly related to the code
```

# Context

# How to trust this PR

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

**Review by Jimbo4350 — APPROVED** [review]
LGTM, devx will need to sort out CI.

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.6.4", "9.8.1"]
+        ghc: ["8.10.7", "9.6.4", "9.8.2", "9.10.1"]
```
Comment (carbolymer) [inline .github/workflows/haskell.yml:23]: do we need to pin the patch version?
```suggestion
        ghc: ["8.10.7", "9.6.4", "9.8.2", "9.10"]
```

File: cardano-api/cardano-api.cabal [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cabal-version: 3.4
 name: cardano-api
-version: 9.1.0.0
+version: 9.2.0.0
```
Comment (carbolymer) [inline cardano-api/cardano-api.cabal:3]: Unnecessary version bump. We're only bumping version when releasing.

**Review by carbolymer — CHANGES_REQUESTED** [review]

File: cardano-api/cardano-api.cabal (hunk 2 above)
Comment (erikd) [reply cardano-api/cardano-api.cabal:3]: Have remove that commit from the PR.

**Review by carbolymer — APPROVED** [review]
LGTM

**Pull request merged**